### PR TITLE
Sync v2-4-stable with v2-4-test to release 2.4.1

### PR DESCRIPTION
**Pull request by jedcunningham**
Time for 2.4.1rc1!

### CODE REVIEW

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -21,6 +21,44 @@
...
+- Log warning when secret backend kwargs is invalid (#26580)
+- Fix grid view log try numbers (#26556)
+- Don't import kubernetes unless you have a V1Pod (#26496)
+- Template rendering issue in passing ``templates_dict`` to task decorator (#26390)
+- Fix Deferrable stuck as ``scheduled`` during backfill (#26205)
+- Suppress SQLALCHEMY_TRACK_MODIFICATIONS warning in db init (#26617)
+- Correctly set ``json_provider_class`` on Flask app so it uses our encoder (#26554)
+- Fix WSGI root app (#26549)
+- Fix deadlock when mapped task with removed upstream is rerun (#26518)
+- ExecutorConfigType should bei ``cacheable`` (#26498)
```
Comment (ephraimbuddy) [inline RELEASE_NOTES.rst]:
```suggestion
- ExecutorConfigType should be ``cacheable`` (#26498)
```

File: RELEASE_NOTES.rst [hunk 2]
```diff
@@ -44,7 +44,7 @@ Bug Fixes
 - Correctly set ``json_provider_class`` on Flask app so it uses our encoder (#26554)
 - Fix WSGI root app (#26549)
 - Fix deadlock when mapped task with removed upstream is rerun (#26518)
-- ExecutorConfigType should bei ``cacheable`` (#26498)
```
Comment (potiuk) [inline RELEASE_NOTES.rst]: VIM :)

Comment (jedcunningham) [reply RELEASE_NOTES.rst]: Oops!

**Review by kaxil — APPROVED** [review]

**Review by dstandish — APPROVED** [review]

**Review by potiuk — APPROVED** [review]
Looked through all changes, all looks good.

**Pull request merged**